rosetta-lang/src/main/java/com/regnosys/rosetta/types/SubtypeRelation.java

```diff
@@ -39,6 +51,15 @@ public boolean isSubtypeOf(RType t1, RType t2) {
 		return false;
 	}
 	
+	public RMetaAnnotatedType join(RMetaAnnotatedType t1, RMetaAnnotatedType t2) {
```

Comment:
Would add an early return here to return `t1` if `t1` equals `t2` - so you don't need to new up new objects.

Reply:
done